backend/infrahub/core/query/node.py

```diff
@@ -851,28 +868,39 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
         )
         self.params.update(branch_params)
 
-        query = """
-        MATCH (n:%(node_kind)s)
-        CALL {
-            WITH n
-            MATCH (root:Root)<-[r:IS_PART_OF]-(n)
+        if (not self.branch.is_default and not self.has_filters) or self.schema.branch == BranchSupportType.AGNOSTIC:
```

Comment:
I'd like to understand why we need the subquery when using filters, it's not super obvious to me...

Reply:
Based on your findings, 
We need the subquery if we are NOT using filters and if we are not in the default branch.
When we are using filters the bad results will be filtered out by the subqueries for the filters.

Reply:
Understood :smiley: 
Could we document that either in code comments or in the commit message or somewhere else for posterity? As you can see it's not obvious even to me who found that out :man_facepalming: 

PS: I think that behavior also exists in the ordering subqueries (done in `build_subquery_order`)

Reply:
I added a comment
also after more testing, even when we are using filters (in a branch), the subquery is required.

Reply:
Thanks, good catch. I was not very sure about that point, thanks for testing (We absolutely need more testing for that subquery...)